shared/src/lib/utils.js

```diff
@@ -0,0 +1,92 @@
+export async function summarizeContent({
+  url,
+  text,
+  summary_type,
+  target_language,
+  api_engine,
+  token,
+  api_token,
+}) {
+  let summary = 'Unknown error';
+  let success = false;
+  const useApi = Boolean(api_token);
+
+  try {
+    const requestParams = {
+      url,
+      summary_type,
+    };
+
+    if (target_language) {
+      requestParams.target_language = target_language;
+    }
+
+    if (api_engine && useApi) {
+      requestParams.engine = api_engine;
+    }
+
+    if (useApi) {
+      if (api_engine) {
+        requestParams.engine = api_engine;
+      }
+
+      if (text) {
+        requestParams.text = text;
+        requestParams.url = undefined;
```

Comment:
TIL [about this rule](https://docs.rome.tools/lint/rules/nodelete/) with using `delete requestParams.url`.